core/blockchain.go

```diff
@@ -1344,6 +1349,12 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 		bc.insert(block)
 	}
 	bc.futureBlocks.Remove(block.Hash())
+
+	if header.ShardID() == shard.BeaconChainShardID {
```

Comment:
This should only be executed during pre-staking time, bascially preStakingEpoch <= Epoch < StakingEpoch